core/src/main/java/com/adobe/aem/commons/assetshare/content/properties/impl/AssetTypeImpl.java

```diff
@@ -70,16 +70,22 @@ public String getLabel() {
     }
 
     @Override
-    public String get(Asset asset) {
-        final ResourceResolver resourceResolver = asset.adaptTo(Resource.class).getResourceResolver();
+    public String get(final Asset asset) {
+        final Resource assetResource = asset.adaptTo(Resource.class);
+        final ResourceResolver resourceResolver = assetResource.getResourceResolver();
         final String dcFormat = StringUtils.defaultIfBlank(asset.getMimeType(), "");
 
         final String ext = StringUtils.defaultIfBlank(dcFormat.substring(dcFormat.lastIndexOf('/') + 1, dcFormat.length()), "");
         final Resource lookedupResource = resourceResolver.getResource("/mnt/overlay/dam/gui/content/assets/jcr:content/mimeTypeLookup");
 
         String displayMimeType = null;
 
-        if (lookedupResource != null && !ResourceUtil.isNonExistingResource(lookedupResource)) {
+        /* handles content fragments */
+        if (null != assetResource.adaptTo(ContentFragment.class)) {
```

Comment:
Once we sort of what computed property-based approach we can use to determining if its a CF... we can do something like [0], get that computed property and use that to perform the check. Ideally we isolate all the "is CF" checks centrally.


[0] https://github.com/Adobe-Marketing-Cloud/asset-share-commons/blob/develop/core/src/main/java/com/adobe/aem/commons/assetshare/content/properties/impl/ResolutionImpl.java#L43